packages/program/src/contract-call-script.ts

```diff
@@ -90,7 +93,7 @@ const getSingleCallInstructions = (
     inst.push(asm.call(0x10, 0x11, 0x12, asm.RegId.cgas().to_u8()));
   }
 
-  if (outputInfo.isHeap) {
+  if (outputInfo.encoding === ENCODING_V0 && outputInfo.isHeap) {
```

Comment:
Only set the heap registers under `v0`, this causes overflow issues under `v1` and is unnecessary as heap data is now treated as other call data types and concatenated.